gui/gui/src/ts/components/search/NodeSearchBox.tsx

```diff
@@ -1,41 +1,68 @@
 import * as React from "react";
+import * as qs from "qs";
 
 import {Paper, InputAdornment, InputBase} from "@material-ui/core";
 
 import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
 import {faSearch} from "@fortawesome/free-solid-svg-icons";
+import {useHistory} from "react-router-dom";
+import {Hrefs} from "Hrefs";
 
 export const NodeSearchBox: React.FunctionComponent<{
-  onSubmit: (value: string) => void;
-}> = ({onSubmit}) => {
-  const [search, setSearch] = React.useState<{text: string}>({text: ""});
+  placeholder?: string;
+  showIcon?: boolean;
+  onSubmit?: (value: string) => void;
+  style?: React.CSSProperties;
+  value?: string;
+  onChange?: (value: string) => void;
+}> = ({onSubmit, showIcon = false, placeholder, style, value, onChange}) => {
+  const [search, setSearch] = React.useState<{text: string}>({
+    text: value || "",
+  });
+
+  const history = useHistory();
+
+  if (!onSubmit) {
+    onSubmit = (text: string) => {
+      if (text.length === 0) return;
+
+      history.push(
+        Hrefs.nodeSearch + qs.stringify({text}, {addQueryPrefix: true})
```

Comment:
I'd prefer the Hrefs take parameters so URLs are produced consistently everywhere, which is the purpose of the central Hrefs class. You could do `Hrefs.nodeSearch(kwds?: {text?: string})` then use qs in Hrefs if kwds is defined.

Reply:
Ah makes sense, changed